cou/utils/juju_utils.py

```diff
@@ -164,21 +151,15 @@ def connected(self) -> bool:
     @property
     def name(self) -> str:
         """Return model name."""
+        if self.connected:
+            return self._model.name
+
         if self._name is None:
-            self._name = self._model.name
+            # pylint: disable=protected-access
+            self._name = self._model._connector.jujudata.current_model(model_only=True)
```

Comment:
This looks like a  little hacky to me. What if we instantiate a new [FileJujuData](https://github.com/juju/python-libjuju/blob/f91507a94d7fd8ca56a50e60416d29afe2c64575/juju/client/jujudata.py#L87) by passing the output of  [juju_config_dir](https://github.com/juju/python-libjuju/blob/master/juju/utils.py#L40)?

I think it's more explicit

Reply:
I agree that it will be more explicit, but the model can be initialized with different JujuData and than you will simply provided different current model. Like this if we define model another way this part will raise an exception.

Reply:
I can see the pros and cons from both approaches, so I don't have a strong opinion here. But for personal preferences, I'm leaning towards Gabriel's suggestion more because accessing a private variable composes more risks of code breakage.